dispatcher/backend/src/routes/requested_tasks/requested_task.py

```diff
@@ -229,15 +225,26 @@ def get(self, session: so.Session, token: AccessToken.Payload):
             worker = dbm.Worker.get(session, worker_name, WorkerNotFound)
             if worker.user.username == token.username:
                 worker.last_seen = getnow()
-                previous_ip = str(worker.last_ip)
-                worker.last_ip = worker_ip
-
-                # flush to DB so that record_ip_change has access to updated IP
-                session.flush()
 
                 # IP changed since last encounter
-                if USES_WORKERS_IPS_WHITELIST and previous_ip != worker_ip:
-                    record_ip_change(worker_name)
+                if str(worker.last_ip) != worker_ip:
+                    logger.info(
+                        f"Worker IP changed detected for {worker_name}: "
+                        f"IP changed from {worker.last_ip} to {worker_ip}"
+                    )
+                    worker.last_ip = worker_ip
+                    # commit explicitely since we are not using an explicit transaction,
+                    # and do it before calling Wasabi so that changes are propagated
+                    # quickly and transaction is not blocking
+                    session.commit()
+                    if constants.USES_WORKERS_IPS_WHITELIST:
+                        try:
+                            record_ip_change(session=session, worker_name=worker_name)
+                        except Exception:
+                            raise HTTPBase(
+                                status_code=HTTPStatus.SERVICE_UNAVAILABLE,
+                                error="Recording IP changes failed",
+                            )
```

Comment:
Shouldn't we commit again after `fond_requested_task_for()` ?

Reply:
I do not expect `find_requested_task_for` to write any data to the DB, so I would prefer not to commit. This is a good safeguard to avoid making unexpected changes to the DB.

Reply:
Indeed ; I tend to forgort that it just offers task and workers initiate the status change 👍